File: specification/specs2/cluster/task_management/list_tasks/ListTasksRequest.ts
class ListTasksRequest extends RequestBase {
	@request_parameter()
	nodes: string[];
	@request_parameter()
	actions: string[];
	@request_parameter()
	detailed: boolean;
	@request_parameter()
	parent_node: string;
	@request_parameter()
	parent_task_id: string;
	@request_parameter()
	wait_for_completion: boolean;
	@request_parameter()
	group_by: GroupBy;
}

File: output/csharp/Cluster/TaskManagement/ListTasks/ListTasksRequest.cs
using Nest.Common;
using Nest.CommonOptions;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Cluster {

	public class ListTasksRequest : RequestBase {
		
		[DataMember(Name="actions")]
		public List<string> Actions { get; set; }


		[DataMember(Name="detailed")]
		public bool Detailed { get; set; }


		[DataMember(Name="group_by")]
		public GroupBy GroupBy { get; set; }


		[DataMember(Name="nodes")]
		public List<string> Nodes { get; set; }


		[DataMember(Name="parent_task_id")]
		public string ParentTaskId { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="wait_for_completion")]
		public bool WaitForCompletion { get; set; }

	}
}
